src/components/calcite-checkbox/calcite-checkbox.tsx

```diff
@@ -177,6 +175,15 @@ export class CalciteCheckbox {
     this.checked = this.initialChecked;
   };
 
+  private nativeLabelClickHandler = (event: MouseEvent): void => {
+    if (!this.el.closest("calcite-label") && (event.target as HTMLElement).nodeName === "LABEL") {
+      const target = event.target as HTMLLabelElement;
+      if (this.el.id && target.htmlFor === this.el.id) {
+        this.toggle();
+      }
```

Comment:
This is mainly required because we suffix the `id` of the hidden input with `-input` so as to distinguish it uniquely in the document from the `calcite-checkbox`'s `id`, so when a native `<label>` targets a `calcite-checkbox` with the `for` attribute, this will ensure the toggle happens properly since the `for` attribute wouldn't target the hidden input (native behavior).